[Cache] Improve OCC page (#12857)
cloudflare/cloudflare-docs: content/cache/concepts/cache-control.md

---
title: Origin Cache Control
pcx_content_type: concept
meta:
  title: Origin Cache Control
---

# Origin Cache Control

> Origin Cache Control is a Cloudflare feature. When enabled on an Enterprise website, it indicates that Cloudflare should strictly respect Cache-Control directives received from the origin server. Free, Pro and Business websites have this feature enabled by default.

Reviewer: Enterprise website OR Enterprise account? If it's an enterprise website, what does that mean exactly?

Reviewer: Also, you've removed the hyphen from the title, but left `Cache-Control` in the text?

---
title: Origin Cache Control
pcx_content_type: concept
meta:
  title: Origin Cache Control
---

# Origin Cache Control

Origin Cache Control is a Cloudflare feature. When enabled on an Enterprise customer's website, it indicates that Cloudflare should strictly respect `Cache-Control` directives received from the origin server. Free, Pro and Business customers have this feature enabled by default.

`Cache-Control` directives in the HTTP response from your origin server provide specific [caching instructions](https://datatracker.ietf.org/doc/html/rfc7234) to intermediary services like Cloudflare.

> With the Origin Cache Control feature enabled, `Cache-Control` directives present in the origin server's response will be followed as specified. For example, if the response includes a `max-age` directive of 3600 seconds, Cloudflare will cache the resource for that duration before checking the origin server again for updates.

Reviewer: ```suggestion
With the Origin Cache Control feature enabled, `Cache-Control` directives present in the origin server's response will be followed as specified. For example, if the response includes a `max-age` directive of 3,600 seconds, Cloudflare will cache the resource for that duration before checking the origin server again for updates.
```

> Using Cloudflare's [Cache Rules](/cache/how-to/cache-rules/) allows users to either augment or override an origin server's `Cache-Control` headers or [default policies](/cache/concepts/default-cache-behavior/) set by Cloudflare.

Reviewer: ```suggestion
Cloudflare's [Cache Rules](/cache/how-to/cache-rules/) allows users to either augment or override an origin server's `Cache-Control` headers or [default policies](/cache/concepts/default-cache-behavior/) set by Cloudflare.
```

- How to enable Origin Cache Control.
- How Origin Cache Control behaves with `Cache-Control` directives.
- How other Cloudflare products interact with `Cache-Control` directives.

## `Cache-control` directives

A `Cache-Control` header can include a number of directives, and the directive dictates who can cache a resource along with how long those resources can be cached before they must be updated.

{{<Aside type="note" header="Note">}}

For more information about `Cache-Control` directives at origin servers, refer to the [Mozilla Cache-Control documentation](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control).

{{</Aside>}}

If multiple directives are passed together, each directive is separated by a comma. If the directive takes an argument, it follows the directive separated by an equal sign. For example: `max-age=86400`.

Directives can be broken down into four groups: [cacheability](/cache/concepts/cache-control/#cacheability), [expiration](/cache/concepts/cache-control/#expiration), [revalidation](/cache/concepts/cache-control/#revalidation), and [other](/cache/concepts/cache-control/#other).

### Cacheability

> Cacheability refers to whether or not a resource should enter a cache, and the directives below indicate a resource’s cacheability.

Reviewer: ```suggestion
Cacheability refers to whether or not a resource should enter a cache, and the directives below indicate a resource's cacheability.
```

- `public` — Indicates any cache may store the response, even if the response is normally non-cacheable or cacheable only within a private cache.
- `private` — Indicates the response message is intended for a single user, such as a browser cache, and must not be stored by a shared cache like Cloudflare or a corporate proxy.
- `no-store` — Indicates any cache, such as a client or proxy cache, must not store any part of either the immediate request or response.

### Expiration

Expiration refers to how long a resource should remain in the cache, and the directives below affect how long a resource stays in the cache.

{{<Aside type="note" header="Note">}}

> Cloudflare respects whichever value is higher: the [Browser Cache TTL](/cache/how-to/edge-browser-cache-ttl/) in Cloudflare or the `max-age` header. You can also simultaneously specify a Cloudflare Edge Cache TTL different than a Browser’s Cache TTL respectively via the `s-maxage` and `max-age` `Cache-Control` headers.

Reviewer: ```suggestion
Cloudflare respects whichever value is higher: the [Browser Cache TTL](/cache/how-to/edge-browser-cache-ttl/) in Cloudflare or the `max-age` header. You can also simultaneously specify a Cloudflare Edge Cache TTL different than a Browser's Cache TTL respectively via the `s-maxage` and `max-age` `Cache-Control` headers.
```

- `no-transform` — Indicates that an intermediary — regardless of whether it implements a cache — must not transform the payload.
- `vary` — Cloudflare does not consider vary values in caching decisions. Nevertheless, vary values are respected when [Vary for images](/cache/advanced-configuration/vary-for-images/) is configured and when the vary header is [`vary: accept-encoding`](/speed/optimization/content/brotli/).
- `immutable` — Indicates to clients the response body does not change over time. The resource, if unexpired, is unchanged on the server. The user should not send a conditional revalidation request, such as `If-None-Match` or `If-Modified-Since`, to check for updates, even when the user explicitly refreshes the page. This directive has no effect on public caches like Cloudflare, but does change browser behavior.

## Enable Origin Cache Control

If you enable Origin Cache Control, Cloudflare will aim to strictly adhere to [RFC 7234](https://datatracker.ietf.org/doc/html/rfc7234).

> Enterprise customers have the ability to select if Cloudflare will adhere to this behavior, enabling or disabling Origin Cache Control for their websites through cache rules in the [dashboard](/cache/how-to/cache-rules/settings/#origin-cache-control-enterprise-only) or via [API](/cache/how-to/cache-rules/settings/#origin-cache-control-enterprise-only). Free, Pro, and Business customers have this option enabled by default and cannot disable it.

Reviewer: Do you want a line between these 2 sentences? As is, line 89 will be rendered in front of line 88
